src/Compilers/CSharp/Portable/Symbols/Source/ModifierUtils.cs

```diff
@@ -215,11 +217,19 @@ public static DeclarationModifiers ToDeclarationModifiers(this SyntaxTokenList m
                 result |= one;
             }
 
-            // the two keywords "protected" and "internal" together are treated as one modifier.
-            if ((result & DeclarationModifiers.AccessibilityMask) == (DeclarationModifiers.Protected | DeclarationModifiers.Internal))
+            switch (result & DeclarationModifiers.AccessibilityMask)
             {
-                result &= ~DeclarationModifiers.AccessibilityMask;
-                result |= DeclarationModifiers.ProtectedInternal;
+                case DeclarationModifiers.Protected | DeclarationModifiers.Internal:
+                    // the two keywords "protected" and "internal" together are treated as one modifier.
+                    result &= ~DeclarationModifiers.AccessibilityMask;
+                    result |= DeclarationModifiers.ProtectedInternal;
+                    break;
+
+                case DeclarationModifiers.Private | DeclarationModifiers.Protected:
```

Comment:
This is an "initial prototype" in the terminology of https://gist.github.com/gafter/c8825a0c889b6a5d6999 , and you can see the .work.md file included in this PR describes the testing scenarios remaining (which is basically each of the specified scenarios). The tests you're asking about would not be classified as among those that are part of the initial prototype.